src/basicmotor.ts

```diff
@@ -0,0 +1,41 @@
+import { Device } from "./device";
+import { Hub } from "./hub";
+
+import * as Consts from "./consts";
+
+import { mapSpeed } from "./utils";
+
+export class BasicMotor extends Device {
+
+
+    constructor (hub: Hub, portId: number, type: number = Consts.DeviceType.UNKNOWN) {
+        super(hub, portId, type);
+    }
+
+
+    /**
+     * Set the motor speed.
+     * @method BasicMotor#setSpeed
+     * @param {number} speed For forward, a value between 1 - 100 should be set. For reverse, a value between -1 to -100. Stop is 0.
+     * @returns {Promise} Resolved upon successful completion of command.
+     */
+    public setSpeed (speed: number) {
```

Comment:
Since basic motors don't have a tachometer, this is actually `setPower`. I guess this makes it easier for the API, but would you then have `setSpeed` and `setPower` for tacho motors or just `setSpeed`?

Reply:
Good point. I guess I haven't liked `power` as I think for most users `speed` is a more appropriate term for what they want, even if it isn't exactly the same. I don't this library should attempt to exactly emulate LWP functionally, instead adding a user friendly layer on top of it.

Perhaps we could have `setSpeed(50)` set power by default, however with an options object (such as `setSpeed(50, { maxPower: 50 })` to switch to maintaining the speed with power?

Reply:
@nutki I've gone with `setPower` and done a 180 on this. :) Will implement `setSpeed` appropriately later.